fe/fe-core/src/main/java/org/apache/doris/catalog/PipelineResourceGroup.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.doris.thrift.TPipelineResourceGroup;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.gson.annotations.SerializedName;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+public class PipelineResourceGroup implements Writable, GsonPostProcessable {
+
+    private static final Logger LOG = LogManager.getLogger(PipelineResourceGroup.class);
+
+    @SerializedName(value = "id")
+    private long id;
+
+    @SerializedName(value = "name")
+    private String name;
+
+    @SerializedName(value = "properties")
+    private Map<String, String> properties;
+
+    // version update required after alter resource group
+    @SerializedName(value = "version")
+    private long version;
+
+    @SerializedName(value = "parentName")
+    private String parentName;
+
+    @SerializedName(value = "childrenNames")
+    private Set<String> childrenNames;
+
+    public PipelineResourceGroup(long id, String name, Map<String, String> properties, String parentName) {
+        this.id = id;
+        this.name = name;
+        this.properties = properties;
+        this.parentName = parentName;
+        this.childrenNames = Sets.newHashSet();
+        this.version = 0;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public String getParentName() {
+        return parentName;
+    }
+
+    public void getProcNodeData(BaseProcResult result) {
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            result.addRow(Lists.newArrayList(String.valueOf(id), name, entry.getKey(), entry.getValue()));
+        }
+    }
+
+    @Override
+    public String toString() {
+        return GsonUtils.GSON.toJson(this);
+    }
+
+    @Override
+    public PipelineResourceGroup clone() {
+        PipelineResourceGroup copied = DeepCopy.copy(this, PipelineResourceGroup.class, FeConstants.meta_version);
+        if (copied == null) {
+            LOG.warn("failed to clone resource group: " + getName());
+            return null;
+        }
+        return copied;
+    }
+
+    public TPipelineResourceGroup toThrift() {
+        return new TPipelineResourceGroup(id, name, properties, version);
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        String json = GsonUtils.GSON.toJson(this);
+        Text.writeString(out, json);
+    }
+
+    public static PipelineResourceGroup read(DataInput in) throws IOException {
+        String json = Text.readString(in);
+        return GsonUtils.GSON.fromJson(json, PipelineResourceGroup.class);
+    }
+
+    @Override
+    public void gsonPostProcess() throws IOException {
```

Comment:
No need to implements `GsonPostProcessable` if not used

Reply:
done